### PR TITLE
Video playback

### PR DESCRIPTION
**Pull request by ethan-riback**
Created a VideoPage that a user can use to view both a video and video information.

Created a custom video player to provide a unique video UI.

Updated upload controller as a bug arouse.

### CODE REVIEW

File: JamJar/Artist.swift [hunk 1]
```diff
@@ -35,8 +36,9 @@ class Artist: NSObject, Mappable {
      */
     func mapping(map: Map) {
         id <- map["id"]
+        spotifyResponseId <- map["id"]
         name <- map["name"]
-        spotify_id <- map["spotify_id"]
+        spotifyId <- map["spotify_id"] // Mark found this issue
```
Comment (markkohdev) [inline JamJar/Artist.swift:41]: lol thanks for the cred :p

Comment (drewbanin) [inline JamJar/Artist.swift:41]: Can i get some credit too? I didn't do anything. Just like being validated

File: JamJar/EnterConcertInformationController.swift [hunk 2]
```diff
@@ -238,7 +270,7 @@ class EnterConcertInformationViewController: BaseViewController, UITextFieldDele
     // Keep track of video URLs being stored
     func callback()
     {
-        self.queue--
+        self.queue -= 1
```
Comment (markkohdev) [inline JamJar/EnterConcertInformationController.swift:239]: Take a look at the **second** answer in this stack overflow jawn - http://stackoverflow.com/questions/11909629/waiting-until-two-async-blocks-are-executed-before-starting-another-block
You could (unlikely, but possible) run into race conditions doing this this way.  I would use the async group method

File: JamJar/JamJarAVPlayerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        self.bottomBar.backgroundColor = bottomBarColor
+        self.view.addSubview(self.bottomBar)
+        
+        //start video
+        loadingIndicatorView.startAnimating()
+        self.player!.play() // Start the playback
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
```
Comment (markkohdev) [inline JamJar/JamJarAVPlayerViewController.swift:55]: I've never seen someone override this method lol, nice

File: JamJar/JamJarAVPlayerViewController.swift [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        self.bottomBar.frame = CGRect(x: 0, y: self.view.frame.height - 30, width: self.view.frame.width, height: 30)
+        self.updateFullScreenButton()
+        self.updateSeekSlider()
+        self.updateBufferIndicator()
+    }
+    
+    func removeObservers() {
+        self.player!.removeTimeObserver(self.timeObserver)
+        timeObserver = nil
+        self.player!.removeObserver(self, forKeyPath: "currentItem.playbackLikelyToKeepUp", context: playbackLikelyToKeepUpContext)
```
Comment (markkohdev) [inline JamJar/JamJarAVPlayerViewController.swift:69]: Just out of curiousity, what is this? lol  Commentz here would be useful

File: JamJar/JamJarAVPlayerViewController.swift [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    func updateFullScreenButton() {
+        fullScreenButton.frame = CGRectMake(self.view.frame.width - 40, 0, 30, 30)
+    }
+    
+    // TimeRemaining Label
+    func createTimeObserver() {
+        timePassedLabel.text = "0:00/0:00" //Serves as a default value incase nothing is returned
+        timePassedLabel.font = UIFont(name: "Muli", size: 14)
+        let timeInterval: CMTime = CMTimeMakeWithSeconds(1.0, 10)
+        timeObserver = self.player?.addPeriodicTimeObserverForInterval(timeInterval, queue: dispatch_get_main_queue()) {
```
Comment (markkohdev) [inline JamJar/JamJarAVPlayerViewController.swift:183]: :+1: for doing this async

File: JamJar/VideoPageViewController.swift [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        // Register the reusable video cell
+        self.suggestedTableView.registerNib(UINib(nibName: "VideoCell", bundle: nil), forCellReuseIdentifier: "VideoCell")
+        self.suggestedTableView.registerNib(UINib(nibName: "JamJarHeaderCell", bundle: nil), forCellReuseIdentifier: "JamJarHeaderCell")
+        
+        self.suggestedTableView.reloadData()
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
```
Comment (markkohdev) [inline JamJar/VideoPageViewController.swift:61]: We should probably actually use this haha

**Comment by markkohdev** [comment]
This PR looks spectacular!  Although I didn't get to run this yet haha. But I assume we'll change a buncha stuff here for JamJars

**Comment by markkohdev** [comment]
Just left a few small comments. One last comment, could you run through this and add more comments to some of the more obscure methods/variables plz :)

**Pull request merged**